aether-couchdb-sync-module/conf/check_vars.sh

```diff
@@ -29,12 +29,15 @@ check_variable() {
 }
 
 # Admin user
-check_variable $ADMIN_USERNAME      "Admin user username (ADMIN_USERNAME)"
-check_variable $ADMIN_PASSWORD      "Admin user password (ADMIN_PASSWORD)"
+check_variable "$ADMIN_USERNAME"      "Admin user username (ADMIN_USERNAME)"
+check_variable "$ADMIN_PASSWORD"      "Admin user password (ADMIN_PASSWORD)"
 
 # Aether kernel
-check_variable $AETHER_KERNEL_URL   "Aether KERNEL url (AETHER_KERNEL_URL)"
-check_variable $AETHER_KERNEL_TOKEN "Aether KERNEL token (AETHER_KERNEL_TOKEN)"
+check_variable "$AETHER_KERNEL_URL"   "Aether KERNEL url (AETHER_KERNEL_URL)"
+check_variable "$AETHER_KERNEL_TOKEN" "Aether KERNEL token (AETHER_KERNEL_TOKEN)"
 
 # Google ID
-check_variable $GOOGLE_CLIENT_ID    "Google Client ID (GOOGLE_CLIENT_ID)"
+check_variable "$GOOGLE_CLIENT_ID"    "Google Client ID (GOOGLE_CLIENT_ID)"
+
+# set default value for DEBUG if missing
+export DEBUG="$DEBUG"
```

Comment:
Looking at how this script is run (inside of e.g. `kernel/entrypoint.sh`), `DEBUG` wouldn't get exported into the `kernel/entrypoint.sh` environment. Parent->child process/shell would work, but not the other way around. Also, I'm not quite sure what this line would be doing anyway -- it seems to just reassign the value of `$DEBUG` to itself, or?

Reply:
My suggestion was that we did this before setting the `-u` flag so that it wouldn't fail:
```bash
DEBUG="$DEBUG"
set -u
# the rest of the script
```
But I agree that here it won't work.

Reply:
What about removing the `-u` flag in the `entrypoint.sh` file? We can check all of the required/used env vars in the `check_vars.sh` script :grimacing: 

```bash 
# App requirements
check_variable "$DB_NAME"           "Database name (DB_NAME)"
check_variable "$WEB_SERVER_PORT"   "Web server port (WEB_SERVER_PORT)"
```

Reply:
We could also keep the `-u` flag in the entrypoints and remove the `check_vars` files? :smile: 

To be clear, I'm perfectly ok with `check_vars.sh`, but the argument would go like this: `-u` is a bit more future-proof -- no need to remember adding new vars to the `check_vars.sh` scripts in all modules.  Result would be the same -- `entrypoint.sh` would exit with `1` whenever it encounters an unset variable. Downside: error message not as verbose; they would look like
```
<filename> <linenumber>: <env-var-name> parameter not set
```
I guess there are a couple of environment variables which are not in the entrypoints but still are required. Those could be checked whenever they are retrieved for the first time (most likely in `settings.py`).